pre_commit/git.py

```diff
@@ -197,6 +198,26 @@ def commit(repo: str = '.') -> None:
     cmd_output_b(*cmd, cwd=repo, env=env)
 
 
+def update_changes_concurrent(repo: str = '.') -> None:
+    # If something else is happening in the git repository (e.g. an editor calling git status) this
+    # command might fail. Retry a few times to see if it will succeed.
+    MAX_TRIES = 3
+    for i in range(MAX_TRIES):
+        try:
+            update_changes(repo)
+            return
+        except CalledProcessError as e:
+            if i + 1 < MAX_TRIES and e.returncode == 128 and e.stderr and b'.git/index.lock' in e.stderr:
+                time.sleep(0.5)
+                continue
+            else:
+                raise
+
+
+def get_editor_script_path(repo: str = '.') -> str:
+    return os.path.join('.', get_git_dir(repo), 'hooks', 'editor')
```

Comment:
We're just kinda "making this path up", right? Eg, unlike everything many other things in this directory this doesn't have any predefined "well known" special meaning? If so, might be good to leave a comment here to that effect.

Reply:
Yeah. Great callout, thanks -- added!